service/controller/v10/adapter/mocks.go

```diff
@@ -7,12 +7,18 @@ import (
 	"github.com/aws/aws-sdk-go/aws/request"
 	awscloudformation "github.com/aws/aws-sdk-go/service/cloudformation"
 	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
 	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/elb/elbiface"
 	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/iam/iamiface"
 	"github.com/aws/aws-sdk-go/service/kms"
+	"github.com/aws/aws-sdk-go/service/kms/kmsiface"
 )
 
 type EC2ClientMock struct {
+	ec2iface.EC2API
```

Comment:
Why is that line needed?

Reply:
This is needed to have `EC2ClientMock` fulfills the `ec2iface.EC2API` interface so we can later on assign `EC2ClientMock` to aws.Clients.EC2 which accept a `ec2iface.EC2API` interface.
https://github.com/giantswarm/aws-operator/pull/933/files#diff-1044bb920626db9f96ba5e53d1c4295cR103

Reply:
In order to fulfill the interface you don't need to embed it. I think that's also the reason you had to remove pointer receivers.

Reply:
To fulfill the Interface I would otherwise have to implement all its methods. Embedding the interface allow to fulfill the interface without having to implements all its methods.

The downside is that calling an method which is not implemented will result in panic, but since we use those mock in our tests we know what we are doing here.

Reply:
I didn't know that. It's interesting.

But you should be able to leave the pointer receivers: https://play.golang.org/p/auB-Dm5cD0a

Reply:
Yes I actually tried this and it worked.